virttest/qemu_vm.py

```diff
@@ -2396,6 +2396,20 @@ def sort_key(dev):
 
         return devices, spice_options
 
+    def _br_tap_cleanup(self, nic):
+        def _del_tap(nic):
+            br_mgr, br_name = utils_net.find_current_bridge(nic.ifname)
+            if br_name == nic.netdst:
+                br_mgr.del_port(nic.netdst, nic.ifname)
+
+        if nic.ifname not in utils_net.get_net_if():
+            _del_tap(nic)
```

Comment:
Looks like it still can go wrong at here or L2409 in case of using bridge.

Reply:
Maybe not?
If the nic is not in `ip link`, `_del_tap` will check it on the database, delete it if it exists, do nothing if it doesn't exist.